### DISCUSSION TITLE
rel=me tag in nextjs 13?

### DISCUSSION THREAD

**Discussion by sunnz** [vercel/next.js#51144] [Help] [answered]
### Summary

What's the best way to add a `<link rel="me" value="example.com" />` tag into the `<head></head>` in NextJS 13?

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/me

The Metadata API's verification field generates a `<meta name="me" content="my-link" />` tag but not a `<link>` tag:

https://nextjs.org/docs/app/api-reference/functions/generate-metadata#verification

I don't imagine you could/should add a rel=me into the `<head></head>` in the root layout for verification backends/bots and search engines?

### Additional information

```js
// app/layout.tsx
import "./globals.css";

export const metadata = {
  title: "Create Next App",
  description: "Generated by create next app",
};

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      <head>
        <link rel="me" href="https://example.com" />
      </head>
      <body>
        {children}
      </body>
    </html>
  );
}
```

**Comment by devjiwonchoi** [comment]
No, it is not necessary to put `rel="me"` on the root layout since the `rel="me"` is used for informing the user that has been accessed from another page that has been created by you.

Also, if you put the `rel="me"` on the root layout, ALL your site pages will have that attribute and might cause a negative effect on SEO.

**Reply by sunnz** [reply]
Hey thanks for pointing that out :)

I do want to add rel=me link tag to just the home page inside the `<head>` for mastodon url verification for example

**Reply by devjiwonchoi** [reply, marked as answer]
If so, you can use `Route Group` and separate the main page to have a different `root layout`.

> To create multiple [root layouts](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required), remove the top-level layout.js file, and add a layout.js file inside each route groups.

Create a folder as `(slug)`, for instance `(home)`. Then add your `HomeLayout` inside the Route Group with your home page.

![Screenshot 2023-06-12 at 1 13 17 PM](https://github.com/vercel/next.js/assets/120007119/2ad59b31-772c-4fe9-aed0-21151f8cdcb3)

```js
// /app/(home)/layout.tsx

import '../globals.css' // escape with ../ since it is inside the Route Group folder

export const metadata = {
  title: 'Create Next App',
  description: 'Generated by create next app',
}

export default function HomeLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return (
    <html lang='en'>
      <head>
        <link rel='me' href='https://example.com' />
      </head>
      <body>{children}</body>
    </html>
  )
}
```

## Reference
- [Creating multiple root layouts](https://nextjs.org/docs/app/building-your-application/routing/route-groups#creating-multiple-root-layouts)

**Reply by sunnz** [reply]
I see, what about every page except for the home page, would I need to create a separate Route Group (eg `(everything)`) and all other pages under it?

```ts
// /app/(everything)/layout.tsx

import '../globals.css' // escape with ../ since it is inside the Route Group folder

export const metadata = {
  title: 'Create Next App',
  description: 'Generated by create next app',
}

export default function EverythingLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return (
    <html lang='en'>
      <body>{children}</body>
    </html>
  )
}
```

**Reply by devjiwonchoi** [reply]
Nope. just leave the previous layout.tsx as it is.

![Screenshot 2023-06-12 at 2 24 48 PM](https://github.com/vercel/next.js/assets/120007119/0885f9a9-c97e-4ae2-a083-bb812d9e9793)

**Reply by sunnz** [reply]
Ah gotcha, `app/(home)/layout.tsx` would be used for the home page while everything else would use `app/layout.tsx`
